manifests/database/postgresql.pp

```diff
@@ -91,7 +91,10 @@
   Boolean $password_sensitive  = false,
   Postgresql::Pg_password_encryption $password_encryption = $puppetdb::params::password_encryption,
 ) inherits puppetdb::params {
-  $port = scanf($database_port, '%i')[0]
+  $port = case $database_port.is_a(String) {
```

Comment:
can't we just set Stdlib::Port::Unprivileged as type?

Reply:
Happy to adjust to Stdlib::Port::Unprivileged to refine usage (1025 - 65535). I can change the data type from `Variant[Stdlib::Port, Pattern[/\A[0-9]+\Z/]]` to just `Stdlib::Port::Unprivileged` however this may be a breaking change for users who define the ports as a String presently. The params.pp default port values are currently Strings. 
https://github.com/puppetlabs/puppetlabs-puppetdb/blob/db5fcf74ad0f532f0a06e3e181c5de249df2cb65/manifests/params.pp#L36

Alternatively I can set it to `Variant[Stdlib::Port::Unprivileged, Pattern[/\A[0-9]+\Z/]]` to retain support for String values whilst confining Integers to unprivileged values.

Any preference?

Reply:
The code block is required to avoid a breaking change. If we remove the code block, a user may supply a String value that will be rejected by the date types set on the postgresql::server class and postgresql::server:db defined type. 
The code block converts the `String` value '5432' inherited from params.pp into an Integer 5432. The postgresql::server and defined type postgresql::server:db expect the `port` to be of type Stdlib::Port. They do not accept a String.

Reply:
It may be worth further restricting the ports from `unprivileged` to just `user` to avoid use of the dynamic port range.

Reply:
@bastelfreak I've further refined the port params to use `user` range. I've also extended data types to all child classes and defined types. Please review again when able.